clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetcherTest.java

```diff
@@ -148,6 +150,33 @@ public boolean matches(ClientRequest request) {
     }
 
     @Test
+    public void testFetchedRecordsRaisesOnSerializationErrors() {
+        ByteArrayDeserializer deserializer = new ByteArrayDeserializer() {
+            @Override
+            public byte[] deserialize(String topic, byte[] data) {
+                throw new SerializationException();
```

Comment:
A quick look shows that other code has access to e.g. `topic` and doesn't include it in the exception message. Seems like having the fields there could help better exception messages.